### PR TITLE
updated axis code to have an integer flag

### PR DESCRIPTION
**Pull request by abhishekpratapa**
updated axis code to have an integer flag for only displaying integers on an axis

### CODE REVIEW

File: schema/axis.js [hunk 1]
```diff
@@ -103,6 +104,7 @@ export default {
           "properties": {
             "axis": {"$ref": "#/defs/guideEncode"},
             "ticks": {"$ref": "#/defs/guideEncode"},
+            "integer": {"$ref": "#/defs/guideEncode"},
```
Comment (jheer) [inline schema/axis.js:107]: There is no `"integer"` sub-group of marks. This line should be deleted.

File: src/parsers/axis.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ export default function(spec, scope) {
   datum = {
     orient: spec.orient,
     ticks:  !!value(spec.ticks, config.ticks),
+    integer:  !!value(spec.integer, config.integer),
```
Comment (jheer) [inline src/parsers/axis.js:28]: This line is unnecessary.

File: src/parsers/axis.js [hunk 3]
```diff
@@ -49,6 +50,7 @@ export default function(spec, scope) {
     scale:  scope.scaleRef(spec.scale),
     extra:  config.tickExtra,
     count:  scope.objectProperty(spec.tickCount),
+    integer: scope.property(spec.integer),
```
Comment (jheer) [inline src/parsers/axis.js:53]: This line does not reference the configuration property. However, I think that is fine, as I don't think we need or want a configuration entry for this.

File: src/config.js [hunk 4]
```diff
@@ -145,6 +145,7 @@ function defaults() {
       labelLimit: 180,
       labelPadding: 2,
       ticks: true,
+      integer: false,
```
Comment (jheer) [inline src/config.js:148]: I don't think we need or want a configuration entry for this. Why would someone want to make _all_ axes integer only?!

Comment (abhishekpratapa) [reply src/config.js:148]: Say someone has their own aggregation for a categorical histogram. Specifying an axis to render only integer values is quite useful when we don't have many counts  say (1-4 counts per category). Currently the tick function renders float values for counts that low. tried to use the "format" tag in scales but it didn't change the tick marks properly. Any suggestions to fix this issue?

**Comment by domoritz** [comment]
I filed an issue with a more general solution: https://github.com/vega/vega-encode/issues/14

**Pull request closed without merging**